code_samples/discounts/src/Command/ManageDiscountsCommand.php

```diff
@@ -0,0 +1,88 @@
+use Ibexa\Contracts\Core\Repository\UserService;
+use Ibexa\Contracts\Discounts\DiscountServiceInterface;
+use Ibexa\Contracts\Discounts\Value\DiscountType;
+use Ibexa\Contracts\Discounts\Value\Struct\DiscountCreateStruct;
+use Ibexa\Contracts\Discounts\Value\Struct\DiscountTranslationStruct;
+use Ibexa\Contracts\DiscountsCodes\DiscountCodeServiceInterface;
+use Ibexa\Contracts\DiscountsCodes\Value\Struct\DiscountCodeCreateStruct;
+use Ibexa\Discounts\Value\DiscountCondition\IsInCurrency;
+use Ibexa\Discounts\Value\DiscountCondition\IsInRegions;
+use Ibexa\Discounts\Value\DiscountCondition\IsProductInArray;
+use Ibexa\Discounts\Value\DiscountRule\FixedAmount;
+use Ibexa\DiscountsCodes\Value\DiscountCondition\IsValidDiscountCode;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class ManageDiscountsCommand extends Command
+{
+    protected static $defaultName = 'discounts:manage';
+
+    private DiscountServiceInterface $discountService;
+
+    private DiscountCodeServiceInterface $discountCodeService;
+
+    private PermissionResolver $permissionResolver;
+
+    private UserService $userService;
+
+    public function __construct(UserService $userSerice, PermissionResolver $permissionResolver, DiscountServiceInterface $discountService, DiscountCodeServiceInterface $discountCodeService)
+    {
+        $this->userService = $userSerice;
+        $this->discountService = $discountService;
+        $this->discountCodeService = $discountCodeService;
+        $this->permissionResolver = $permissionResolver;
+
+        parent::__construct();
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $this->permissionResolver->setCurrentUserReference($this->userService->loadUserByLogin('admin'));
+
+        $now = new \DateTimeImmutable();
+
+        $discountCodeCreateStruct = new DiscountCodeCreateStruct(
+            'summer10',
+            null, // Unlimited usage
+            $this->permissionResolver->getCurrentUserReference()->getUserId(),
+            $now
+        );
+        $discountCode = $this->discountCodeService->createDiscountCode($discountCodeCreateStruct);
+
+        $discountCreateStruct = new DiscountCreateStruct();
+        $discountCreateStruct->setIdentifier('discount_identifier')
+            ->setType(DiscountType::CART)
+            ->setPriority(10)
+            ->setEnabled(true)
+            ->setUser($this->userService->loadUserByLogin('admin'))
+            ->setRule(new FixedAmount(10))
+            ->setStartDate($now)
+            ->setConditions([
+                new IsInRegions(['germany', 'france']),
+                new IsProductInArray(['product-1', 'product-2']),
+                new IsInCurrency('EUR'),
+                new IsValidDiscountCode($discountCode->getCode(), $discountCode->getUsedLimit()),
```

Comment:
TBH I'm not sure about the `$discountCode->getUsedLimit()` part - which value should be passed here, the current usage or the max usage?

If I understand correctly, the [condition](https://github.com/ibexa/discounts-codes/blob/4.6/src/lib/Value/DiscountCondition/IsValidDiscountCode.php#L65) calls the [resolver](https://github.com/ibexa/discounts-codes/blob/4.6/src/lib/ExpressionLanguage/ExpressionProvider/IsValidDiscountCodeResolver.php#L36) which takes a single argument - so the second argument to IsValidDiscountCode is actually not used?

Reply:
Maybe @Steveb-p can clarify, but to me it's the max usage.